utils/datasets/cifar10_dataset.py

```diff
@@ -0,0 +1,343 @@
+TEST_SAMPLES_PER_CLIENT = 1000
+NUM_CLASSES = 10
+
+def load_cifar10_federated(
+    num_clients: int = 10, 
+    nit: float = 1, 
+    train_client_batch_size: int = 20, 
+    test_client_batch_size: int = 100):
+  '''Construct a synthetic federated CIFAR-10 from its centralized version
+  Loads the train dataset into a non iid distribution over clients using the
+  sampling method based on LDA, taken from this paper:
+  Measuring the Effects of Non-Identical Data Distribution for
+  Federated Visual Classification (https://arxiv.org/pdf/1909.06335.pdf).
+  Args:
+    num_clients: An integer specifing the total number of clients.
+    nit: A float parametrizing the Dirichlet distribution which samples a 
+      multinomial over classes for each client.
+    train_client_batch_size: A float representing the batch size during 
+      training.
+    test_client_batch_size: A float representing the batch size during
+      test.
+  Returns:
+    A tuple of `tff.simulation.ClientData` representing unpreprocessed
+    train data and test data.
+  '''
+  (train_images, train_labels), (test_images, test_labels) \
+      = datasets.cifar10.load_data()
+  train_clients = collections.OrderedDict()
+  test_clients = collections.OrderedDict()
+
+  proportion_clients_train = []
+  proportion_clients_test = []
+  # each column is a distribution over classes for the client at that index
+  for i in range(num_clients):
+      proportion = np.random.dirichlet(nit*np.ones(NUM_CLASSES,))
+      train_proportion = (5000*proportion).astype(int)
+      proportion_clients_train.append(train_proportion)
+      test_proportion = (1000*proportion).astype(int)
+      proportion_clients_test.append(test_proportion)
+  
+  train_idx_batch = [[] for _ in range(num_clients)]
+  test_idx_batch = [[] for _ in range(num_clients)]
+  proportion_clients_train = np.array(proportion_clients_train)
+  proportion_clients_test = np.array(proportion_clients_test)
+
+  cumsum_client_labels_train = proportion_clients_train.cumsum(axis=0)
+  cumsum_client_labels_test = proportion_clients_test.cumsum(axis=0)
+
+  for k in range(NUM_CLASSES):
+    train_label_k = np.where(train_labels==k)[0]
+    np.random.shuffle(train_label_k)
+    train_split_labels = np.split(train_label_k, cumsum_client_labels_train[:, k])
+    train_idx_batch = [idx_j + splitk.tolist() for idx_j, splitk \
+       in zip(train_idx_batch, train_split_labels)]
+    
+    test_label_k = np.where(test_labels==k)[0]
+    np.random.shuffle(test_label_k)
+    test_split_labels = np.split(test_label_k, cumsum_client_labels_test[:, k])
+    test_idx_batch = [idx_j + splitk.tolist() for idx_j, splitk \
+       in zip(test_idx_batch, test_split_labels)]
+
+  for i in range(num_clients):
+    client_name = str(i)
+    x_train = train_images[np.array(train_idx_batch[i])]
+    y_train = train_labels[np.array(train_idx_batch[i])].astype('int64').squeeze()
+    train_samples_per_client = (x_train.shape[0] \
```

Comment:
The dependency on `client_batch_size` is strange. Add some comments if it is absolutely necessary.